src/ganj/AIGuesser.java

```diff
@@ -0,0 +1,37 @@
+package ganj;
+
+import java.util.Random;
+import java.util.Scanner;
+
+import static ganj.CompareResult.GREATER;
+import static ganj.CompareResult.LOWER;
+
+public class AIGuesser implements Guesser {
+    private int from;
+    private int to;
+    private int guessByAI;
+
+    @Override
+    public void initialize(int from, int to) {
+        this.from = from;
+        this.to = to;
+    }
+
+    @Override
+    public int makeGuess() {
+        return guessByAI = (from + to) / 2;
```

Comment:
Great strategy!

Minor: I propose to split this line in two parts:
first - calculate a guess and assign a variable
second - return variable value